internal/codespaces/api/api.go

```diff
@@ -188,36 +189,54 @@ type CodespaceEnvironmentConnection struct {
 
 // ListCodespaces returns a list of codespaces for the user.
 func (a *API) ListCodespaces(ctx context.Context) ([]*Codespace, error) {
-	req, err := http.NewRequest(
-		http.MethodGet, a.githubAPI+"/user/codespaces", nil,
-	)
-	if err != nil {
-		return nil, fmt.Errorf("error creating request: %w", err)
-	}
+	page := 1
+	per_page := 50
+	codespaces := []*Codespace{}
 
-	a.setHeaders(req)
-	resp, err := a.do(ctx, req, "/user/codespaces")
-	if err != nil {
-		return nil, fmt.Errorf("error making request: %w", err)
-	}
-	defer resp.Body.Close()
+	for {
+		req, err := http.NewRequest(
+			http.MethodGet, a.githubAPI+"/user/codespaces", nil,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("error creating request: %w", err)
+		}
+		a.setHeaders(req)
+		q := req.URL.Query()
+		q.Add("page", strconv.Itoa(page))
+		q.Add("per_page", strconv.Itoa(per_page))
 
-	b, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return nil, fmt.Errorf("error reading response body: %w", err)
-	}
+		req.URL.RawQuery = q.Encode()
+		resp, err := a.do(ctx, req, "/user/codespaces")
+		if err != nil {
+			return nil, fmt.Errorf("error making request: %w", err)
+		}
+		defer resp.Body.Close()
```

Comment:
Defer in loops are usually bad. Perhaps it's better to create a new inline function. Or a new method.

ie

```
fetchCodespaces := func() ([]*Codespace, error) {
  ...
  defer resp.BodyClose()
}

...

for {
   codespaces, err := fetchCodespaces()
}
```